source/frontend/src/components/maps/MapLeafletView.tsx

```diff
@@ -218,10 +219,21 @@ const MapLeafletView: React.FC<React.PropsWithChildren<MapLeafletViewProps>> = (
 
         <LegendControl />
         <ZoomOutButton />
+        <AdvancedMapFilterButton
+          open={mapMachine.isAdvancedFilterSidebarOpen}
+          onOpen={() => {
+            setLayersOpen(false);
+            mapMachine.openAdvancedFilterSidebar();
+          }}
+          onClose={() => mapMachine.closeAdvancedFilterSidebar()}
```

Comment:
Let's move this logic to the map FSM given it is coordinating between different states

Reply:
SetLayersOpen is **not** in the FSM - it is on local state. I'd rather do that on a separate story so we can have the filter sidebar merged

Reply:
Take a look at the PR I posted, it controls the visibility directly on the fsm

Reply:
My two cents, prefer having the FSM manage the state of the advanced search/sidebar. I wouldn't say that because the layer is currently controlled by local state that it should continue to be, I'd rather have a consistent implementation.